topo_processor/metadata/metadata_loaders/metadata_loader_imagery_historic.py

```diff
@@ -192,3 +198,44 @@ def add_datetime_property(self, item: Item, asset_metadata: Dict[str, str]):
                 item.add_error(msg="Invalid date", cause=self.name, e=e)
         else:
             item.add_error(msg="No date found", cause=self.name, e=Exception(f"item date has no value"))
+
+    def add_centroid(self, item: Item, asset_metadata: Dict[str, str]):
+
+        centroid = {
+            "lat": string_to_number(asset_metadata.get("photocentre_lat", None)),
+            "lon": string_to_number(asset_metadata.get("photocentre_lon", None)),
+        }
+        if self.is_valid_centroid(item, centroid):
+            item.properties["proj:centroid"] = centroid
+            item.add_extension(stac.StacExtensions.projection.value)
+
+    def is_valid_centroid(self, item: Item, centroid) -> bool:
+        if not all(centroid.values()) or "" in centroid.values():
+            item.add_warning(
+                msg="Skipped Record",
+                cause=self.name,
+                e=Exception(f"stac field 'proj:centroid' has None values: {centroid}"),
+            )
+            return False
+        if any(isinstance(value, str) for value in centroid.values()):
+            item.add_warning(
+                msg="Skipped Record",
+                cause=self.name,
+                e=Exception(f"stac field 'proj:centroid' has invalid lat value: {centroid['lat']}"),
```

Comment:
This entire function could be simplified to:

```python
if not isinstance(centeroid["lat"], numbers.Number) or centeroid["lat"] > 90 or centeroid["lat"] < -90:
  # lat invalid
  
if not isinstance(centeroid["lon"], numbers.Number) or centeroid["lon"] > 180 or centeroid["lon"] < -180:
  # lon invalid
```

The `numbers.Number` matches both `int` and `float` so will exclude None, str, Dict etc..

It would be great to have a few unit tests for this function.